modules/otel-config-instrumentation.adoc

```diff
@@ -8,11 +8,25 @@
 
 The {OTELName} can inject and configure the OpenTelemetry auto-instrumentation libraries into your workloads. Currently, the project supports injection of the instrumentation libraries from Go, Java, Node.js, Python, .NET, and the Apache HTTP Server (`httpd`).
 
-Auto-instrumentation in OpenTelemetry refers to the capability where the framework automatically instruments an application without manual code changes. This enables developers and administrators to get observability into their applications with minimal effort and changes to the existing codebase.
+== How Autoinstrumentation works in the OpenTelemetry Operator
+
+Auto-instrumentation in {OTELOperator} refers to the capability where the framework automatically instruments an application without manual code changes. This enables developers and administrators to get observability into their applications with minimal effort and changes to the existing codebase.
+
+This process works as follows:
+
+1. **Container Modification**: The OpenTelemetry Operator modifies the container definition of your application. It adds the necessary instrumentation libraries for the specific language your application is written in (e.g., Java, Python, .NET, Node.js). This happens creating an init-container.
```

Comment:
Question:
is it possible to be more specific about what the `container definition` (of your application) is? What kind of object are you talking about?

Reply:
When the instrumentation is injected, this happens:
- One init-container is added to the pod (all the languages except Go)
- (For go) a sidecar container is injected in that pod

This is what this paragraph tries to explain in a generic way.

It is something that modifies the application pods.

Reply:
WDYT?
```
1. The {OTELOperator} adds the necessary instrumentation libraries for the programming language of the instrumented application through injection of an init-container, or as in the case of Go, a sidecar container.
```

Reply:
WDYT?
```suggestion
1. The {OTELOperator} adds the necessary instrumentation libraries for the programming language of the instrumented application through injection of an init-container, or as in the case of Go, a sidecar container.
```

Reply:
WDYT?
```suggestion
1. The {OTELOperator} injects an init-container, or in the case of Go, a side container, to add the necessary instrumentation libraries for the programming language of the instrumented application.
```

Reply:
Done.